### PR TITLE
Avoid triggering IResourceDelta.ENCODING change if encoding is already set to the good value

### PR DESCRIPTION
**Pull request by RoiSoleil**
Avoid triggering IResourceDelta.ENCODING change if encoding is already set to the good value

### CODE REVIEW

File: org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java [hunk 1]
```diff
@@ -488,8 +488,10 @@ private void updateProjectConfiguration(ProjectConfigurationRequest request, IPr
     MavenProject mavenProject = request.mavenProject();
     Properties mavenProperties = mavenProject.getProperties();
     String sourceEncoding = mavenProperties.getProperty("project.build.sourceEncoding");
-    log.debug("Setting encoding for project {}: {}", project.getName(), sourceEncoding); //$NON-NLS-1$
-    project.setDefaultCharset(sourceEncoding, monitor);
+    if(project.getDefaultCharset() == null || !project.getDefaultCharset().equals(sourceEncoding)) {
```
Comment (HannesWell) [inline org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java:491]:
```suggestion
    if(Objects.equals(project.getDefaultCharset(), sourceEncoding)) {
```
That should also cover the case that the project's encoding is already null and no encoding is specified and would resetting it to null again. Furthermore the project's charset is then only fetched once.

If you think that's fine it should be applied to all other changes as well.

**Review by HannesWell — CHANGES_REQUESTED** [review]
Thank you for this contribution.
In general this looks sane, but I think the implementation can be slightly enhanced.

File: org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java (hunk 1 above)
Comment (RoiSoleil) [reply org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java]: Done ;)

**Review by HannesWell — APPROVED** [review]
Looks good, thank you. :)

**Comment by thahnen** [comment]
Thanks a lot, @RoiSoleil and @HannesWell, for fixing this issue! Much appreciated :bow:

**Pull request merged**